lib/msf/util/exe.rb

```diff
@@ -2258,6 +2262,10 @@ def self.elf?(code)
     code[0..3] == "\x7FELF"
   end
 
+  def self.macho?(code)
+    code[0..3] == "\xCF\xFA\xED\xFE"
```

Comment:
To prevent weird bugs in the future, this also needs to check for the 32-bit magic (`\xCE\xFA\xED\xFE`), the big-endian 32-bit and 64-bit versions of the magics, and the fat-binary magic (`\xCA\xFE\xBA\xBE`).